[CoreIPC] Messages::GPUConnectionToWebProcess::SetUserPreferredLanguages is sent from WCP and not UIP

cdumez opened the pull request:
#### cfd1105a34372cb245b99f959ab10c8bb2a1a48d
<pre>
[CoreIPC] Messages::GPUConnectionToWebProcess::SetUserPreferredLanguages is sent from WCP and not UIP
<a href="https://bugs.webkit.org/show_bug.cgi?id=256014">https://bugs.webkit.org/show_bug.cgi?id=256014</a>
&lt;rdar://107917551&gt;

Reviewed by Alex Christensen.

The UIProcess normally knows about language overrides and is thus able to IPC
them to the GPUProcess. This is to avoid having the untrusted WebProcesses
send the languages to the GPUProcess themselves.

Note that there is one exception: internals.setUserPreferredLanguages(). Since
this is called in the WebProcess, tests calling this will only override the
languages in the current WebProcess, not more globally (e.g. in the GPUProcess).

Tests that want/need to override the language more globally now have to use the
pre-existing `&lt;!-- webkit-test-runner [ language=foo,bar ] --&gt;` test option.

* LayoutTests/http/tests/media/hls/hls-audio-tracks-locale-selection-expected.txt:
* LayoutTests/http/tests/media/hls/hls-audio-tracks-locale-selection.html:
* Source/WebCore/testing/Internals.idl:
* Source/WebKit/GPUProcess/GPUConnectionToWebProcess.cpp:
(WebKit::m_routingArbitrator):
(WebKit::GPUConnectionToWebProcess::setUserPreferredLanguages): Deleted.
* Source/WebKit/GPUProcess/GPUConnectionToWebProcess.h:
* Source/WebKit/GPUProcess/GPUConnectionToWebProcess.messages.in:
* Source/WebKit/GPUProcess/GPUProcess.cpp:
(WebKit::GPUProcess::initializeGPUProcess):
(WebKit::GPUProcess::userPreferredLanguagesChanged):
* Source/WebKit/GPUProcess/GPUProcess.h:
* Source/WebKit/GPUProcess/GPUProcess.messages.in:
* Source/WebKit/GPUProcess/GPUProcessCreationParameters.cpp:
(WebKit::GPUProcessCreationParameters::encode const):
(WebKit::GPUProcessCreationParameters::decode):
* Source/WebKit/GPUProcess/GPUProcessCreationParameters.h:
* Source/WebKit/Shared/GPUProcessConnectionParameters.h:
* Source/WebKit/Shared/GPUProcessConnectionParameters.serialization.in:
* Source/WebKit/UIProcess/GPU/GPUProcessProxy.cpp:
(WebKit::GPUProcessProxy::GPUProcessProxy):
* Source/WebKit/UIProcess/WebProcessPool.cpp:
(WebKit::WebProcessPool::setOverrideLanguages):
* Source/WebKit/WebProcess/GPU/GPUProcessConnection.cpp:
(WebKit::getGPUProcessConnectionParameters):
(WebKit::GPUProcessConnection::GPUProcessConnection):
(WebKit::GPUProcessConnection::~GPUProcessConnection):
(WebKit::languagesChanged): Deleted.

Canonical link: <a href="https://commits.webkit.org/263465@main">https://commits.webkit.org/263465@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1147db33c4070a2897d9c34eea030ec6e5069c9b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/4614 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/4733 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/4887 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/6107 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/4769 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/4605 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/4873 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/4693 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/5009 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/4683 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/4774 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/4133 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/6117 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/2262 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/4125 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/9116 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/4122 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/4199 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/5746 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/4591 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/3739 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/4120 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/4121 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1150 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/8163 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/4481 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebCore/testing/Internals.idl:
```diff
@@ -521,6 +521,11 @@ typedef (FetchRequest or FetchResponse) FetchObject;
+    // To override the language more globally, please add something like this to
+    // your test instead:
+    // <!-- webkit-test-runner [ language=jp,fr ] -->
     undefined setUserPreferredLanguages(sequence<DOMString> languages);
```
Can we completely remove this in favor of the html comment in the first line?

cdumez replied:
Maybe, if the tests don't dynamically need to change between several languages. There is a decent amount of tests using this so I didn't attempt in this patch. You're right though, it is something we should try to do. I'll try and follow up.

achristensen07 reviewed (commented):
Why does the GPU Process need to know the languages?
What would this break if we removed all the calls to overrideUserPreferredLanguages?

cdumez committed:
[CoreIPC] Messages::GPUConnectionToWebProcess::SetUserPreferredLanguages is sent from WCP and not UIP
https://bugs.webkit.org/show_bug.cgi?id=256014
<rdar://107917551>

Reviewed by Alex Christensen.

The UIProcess normally knows about language overrides and is thus able to IPC
them to the GPUProcess. This is to avoid having the untrusted WebProcesses
send the languages to the GPUProcess themselves.

Note that there is one exception: internals.setUserPreferredLanguages(). Since
this is called in the WebProcess, tests calling this will only override the
languages in the current WebProcess, not more globally (e.g. in the GPUProcess).

Tests that want/need to override the language more globally now have to use the
pre-existing `<!-- webkit-test-runner [ language=foo,bar ] -->` test option.

* LayoutTests/http/tests/media/hls/hls-audio-tracks-locale-selection-expected.txt:
* LayoutTests/http/tests/media/hls/hls-audio-tracks-locale-selection.html:
* Source/WebCore/testing/Internals.idl:
* Source/WebKit/GPUProcess/GPUConnectionToWebProcess.cpp:
(WebKit::m_routingArbitrator):
(WebKit::GPUConnectionToWebProcess::setUserPreferredLanguages): Deleted.
* Source/WebKit/GPUProcess/GPUConnectionToWebProcess.h:
* Source/WebKit/GPUProcess/GPUConnectionToWebProcess.messages.in:
* Source/WebKit/GPUProcess/GPUProcess.cpp:
(WebKit::GPUProcess::initializeGPUProcess):
(WebKit::GPUProcess::userPreferredLanguagesChanged):
* Source/WebKit/GPUProcess/GPUProcess.h:
* Source/WebKit/GPUProcess/GPUProcess.messages.in:
* Source/WebKit/GPUProcess/GPUProcessCreationParameters.cpp:
(WebKit::GPUProcessCreationParameters::encode const):
(WebKit::GPUProcessCreationParameters::decode):
* Source/WebKit/GPUProcess/GPUProcessCreationParameters.h:
* Source/WebKit/Shared/GPUProcessConnectionParameters.h:
* Source/WebKit/Shared/GPUProcessConnectionParameters.serialization.in:
* Source/WebKit/UIProcess/GPU/GPUProcessProxy.cpp:
(WebKit::GPUProcessProxy::GPUProcessProxy):
* Source/WebKit/UIProcess/WebProcessPool.cpp:
(WebKit::WebProcessPool::setOverrideLanguages):
* Source/WebKit/WebProcess/GPU/GPUProcessConnection.cpp:
(WebKit::getGPUProcessConnectionParameters):
(WebKit::GPUProcessConnection::GPUProcessConnection):
(WebKit::GPUProcessConnection::~GPUProcessConnection):
(WebKit::languagesChanged): Deleted.

Canonical link: https://commits.webkit.org/263465@main
